src/librustc_mir/hair/cx/expr.rs

```diff
@@ -96,6 +96,14 @@ fn apply_adjustment<'a, 'gcx, 'tcx>(cx: &mut Cx<'a, 'gcx, 'tcx>,
             ExprKind::Cast { source: expr.to_ref() }
         }
         Adjust::Deref(None) => {
+            // Not needed in the next match arm since deref coercions always
+            // start with a built-in deref.
```

Comment:
I find this comment sort of confusing. I think you are saying "this adjustment below is not needed in the next arm"? If so, I think it'd be helpful to start with an explanation of what this `if` is actually *doing*. (Ideally, including a Rust example)